docs/core/integration-quality-scale/rules/runtime-data.md

```diff
@@ -46,7 +46,8 @@ More information about configuration entries and their lifecycle can be found in
 
 ## Exceptions
 
-There are no exceptions to this rule.
+Runtime data was previously stored in the `hass.data` untyped dictionary, and checks have been added to ensure this is no longer the case.
+If an integration still needs to access `hass.data` directly, the rule should be marked as `exempt` rather than `done` with a corresponding comment.
```

Comment:
> All integrations will still access `hass.data` for their platforms.

As far as I can tell, they do not access it directly. They only access it via helpers and the implementation is hidden from the integration.

Reply:
Sure, but the point is, we want people to use this correctly. This is basically adding a check saying that using hass.data isn't correct; which simply isn't true?

Also, this rule is not about hass.data at all, it is about use of runtime_data.

Reply:
> Also, this rule is not about hass.data at all, it is about use of runtime_data.

It's more a side-effect of the `runtime_data`, and an easy check to add to IQS validation to relieve pressure on reviewers.
But maybe it is to necessary right now and it can always be added later if there is a need...